### PR TITLE
fix(argo-cd): hard coded .selector.matchLabels

### LINKED ISSUE #602: [argo-cd] Cannot Create 2 helm releases within the same cluster - selector does not match template labels
**Describe the bug**
Resources with cluster scope (ClusterRole) does not have unique name per helm release.
If you try more than one helm release using the same helm chart argo/argo-cd, this will not work

**To Reproduce**
1. Create first helm release in namespace `argocd`
2. Try Creating Second helm release in another namespace `argocd-staging`
3. See error
```log
Error: rendered manifests contain a resource that already exists. Unable to continue with install:
 ClusterRole "argocd-application-controller" in namespace "" 
exists and cannot be imported into the current release: 
invalid ownership metadata;
annotation validation error: key "meta.helm.sh/release-namespace" must equal "argocd-staging": current value is "argocd"
```
4. I assign another value for `.Values.controller.name` and `.Values.server.name` . However, i fall into another error 

```
argocd-staging-application-controller-stagging is invalid: spec.template.metadata.labels: invalid value .....
... selector does not match template labels
```

**Expected behavior**
From one helm chart, many helm releases can be created within the same cluster.

**Screenshots**

**Additional context**

- helm chart name: `argo/argo-cd`
- helm chart version: `2.14.6`

### PR DESCRIPTION
**Pull request by abdennour**
it fixes #602

Checklist:

* [x] I have updated the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [x] I have signed the CLA and the build is green.
* [x] I will test my changes again once merged to master and published.

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -4,7 +4,7 @@
 /charts/argo @stefansedich @paguos @vladlosev @yann-soubeyrand
 
 # Argo CD
-/charts/argo-cd @seanson @spencergilbert @davidkarlsen @mr-sour @yann-soubeyrand
+/charts/argo-cd @seanson @spencergilbert @davidkarlsen @mr-sour @yann-soubeyrand @abdennour
```
Comment (seanson) [inline CODEOWNERS:7]: `CODEOWNER` changes are done typically by one of the ArgoCD owners (such as https://github.com/argoproj/argo-helm/pull/551). if you'd like to volunteer to be a contributor please give a shout in the #argo-helm Slack channel.

**Review by seanson — CHANGES_REQUESTED** [review]
Looks good but needs the CODEOWNERS change removed.

File: charts/argo-cd/templates/argocd-application-controller/deployment.yaml [hunk 2]
```diff
@@ -14,7 +14,7 @@ metadata:
 spec:
   selector:
     matchLabels:
-      app.kubernetes.io/name: {{ include "argo-cd.name" . }}-application-controller
+      app.kubernetes.io/name: {{ include "argo-cd.name" . }}-{{ .Values.controller.name }}
```
Comment (vladlosev) [inline charts/argo-cd/templates/argocd-application-controller/deployment.yaml:17]: It looks like this change will break upgrades of the chart as the `selector` field is [immutable](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#label-selector-updates).

Comment (seanson) [reply charts/argo-cd/templates/argocd-application-controller/deployment.yaml:17]: Given the current value is `application-controller` then this shouldn't break existing upgrades but a README.md callout would be good.

Comment (abdennour) [reply charts/argo-cd/templates/argocd-application-controller/deployment.yaml:17]: Yep! already the default value of `.Values.controller.name` is `application-controller`

File: CODEOWNERS (hunk 1 above)
Comment (abdennour) [reply CODEOWNERS]: > Looks good but needs the CODEOWNERS change removed.
Done! I removed it.
@seanson @vladlosev  kindly proceed

File: CODEOWNERS [hunk 3]
```diff
@@ -4,7 +4,7 @@
 /charts/argo @stefansedich @paguos @vladlosev @yann-soubeyrand
 
 # Argo CD
-/charts/argo-cd @seanson @spencergilbert @davidkarlsen @mr-sour @yann-soubeyrand @abdennour
+/charts/argo-cd @seanson @spencergilbert @davidkarlsen @mr-sour @yann-soubeyrand
```
Comment (abdennour) [inline CODEOWNERS]: @seanson   this is done as per reviewer's request

**Comment by abdennour** [comment]
@seanson  requested changes. And i applied these changes.
What's pending now ?

**Review by seanson — APPROVED** [review]

**Comment by seanson** [comment]
Looks good, thanks @abdennour!

**Pull request merged**